### ISSUE TITLE
Building without default features doesn't work

### ISSUE THREAD

**Issue by beatgammit** [ryanisaacg/quicksilver#338]
I tried building the image example without any default features and I got compile errors.

The following seem necessary to get a simple compile working: fonts, saving, sounds. It looks like `FontError`, `SoundError`, and `SerdeError` are used in `error.rs` and the build fails when they're not defined.

There should be a CI test for building without any features defined to catch stuff like this.

If my experiments with this library work out, I might be able to contribute a fix for this.

**Comment by beatgammit** [comment]
It would also be nice if `serde` and `serde_derive` were optional dependencies as well.

**Comment by beatgammit** [comment]
I get WASM build errors when providing only the above dependencies (gamepad seems to not be optional with WASM builds).

**Comment by ryanisaacg** [comment]
Could you post the build errors, please? Gamepad should be optional with WASM builds.

I agree that managing the optional features in CI is a good idea, I'll see about adding a Travis task that runs a test build with no features enabled.

**Comment by beatgammit** [comment]
Sure:

```
error[E0201]: duplicate definitions with name `provide_gamepads_impl`:
   --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/input/gamepad.rs:132:5
    |
107 | /     fn provide_gamepads_impl(&self, buffer: &mut Vec<Gamepad>) {
108 | |         fn new(id: u32) -> Gamepad {
109 | |             Gamepad { id, buttons: [ButtonState::NotPressed; 17], axes: [0.0; 4] }
110 | |         }
...   |
128 | |         }
129 | |     }
    | |_____- previous definition of `provide_gamepads_impl` here
...
132 | /     fn provide_gamepads_impl(&self, _buffer: &mut Vec<Gamepad>) {
133 | |         //Inentionally a no-op
134 | |     }
    | |_____^ duplicate definition

error: aborting due to previous error

For more information about this error, try `rustc --explain E0201`.
error: Could not compile `quicksilver`.
```

And my `Cargo.toml`:

    quicksilver = { version = "0.2.1", default-features = false, features = ["fonts", "saving", "sounds"]  }

If I disable all default features, desktop builds don't work:

```
error[E0599]: no variant named `SoundError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:46:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `SoundError` not found here
...
46 |             &QuicksilverError::SoundError(ref err) => err.description(),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error[E0599]: no variant named `SerdeError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:48:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `SerdeError` not found here
...
48 |             &QuicksilverError::SerdeError(ref err) => err.description(),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error[E0599]: no variant named `FontError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:49:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `FontError` not found here
...
49 |             &QuicksilverError::FontError(ref err) => err.description()
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error[E0599]: no variant named `SoundError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:57:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `SoundError` not found here
...
57 |             &QuicksilverError::SoundError(ref err) => Some(err),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error[E0599]: no variant named `SerdeError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:59:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `SerdeError` not found here
...
59 |             &QuicksilverError::SerdeError(ref err) => Some(err),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error[E0599]: no variant named `FontError` found for type `error::QuicksilverError` in the current scope
  --> /home/otto/.cargo/registry/src/github.com-1ecc6299db9ec823/quicksilver-0.2.1/src/error.rs:60:14
   |
20 | pub enum QuicksilverError {
   | ------------------------- variant `FontError` not found here
...
60 |             &QuicksilverError::FontError(ref err) => Some(err)
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `error::QuicksilverError`

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `quicksilver`.

To learn more, run the command again with --verbose.
```

**Comment by beatgammit** [comment]
I'm mostly interested in excluding those features to improve compile times by reducing dependencies. I expect that most of that stuff gets optimized away by the compiler, but I don't know enough about the WASM compiler to know if that's a thing.

**Comment by ryanisaacg** [comment]
This should be resolved by the #341, if it's not let me know and I'll reopen
